markdowns/20_REST_API.md

```diff
@@ -0,0 +1,370 @@
+)
+```
+
+A document search pipeline requires a DocumentStore and a Retriever. Use `ElasticsearchDocumentStore` and `EmbeddingRetriever` for these nodes respectively and define them under `components`. For each component, `type` needs to refer to a pipeline node in Haystack, `name` refers how these nodes are called in the pipeline YAML, and `params` is used for the parameters that can be provided. As Retriever parameters, provide `document_store`, `embedding_model` and a `top_k` value. 
+
+```yaml
+components:
+  - name: DocumentStore
+    type: ElasticsearchDocumentStore
+  - name: Retriever
+    type: EmbeddingRetriever
+    params:
+      document_store: DocumentStore
+      top_k: 5 
+      embedding_model: sentence-transformers/multi-qa-mpnet-base-dot-v1
+```
+
+After defining components, create a query pipeline in `pipelines` section. Here, `name` refers to the name of the pipeline and `nodes` defines how the pipeline is built. 
+
+```yaml
+pipelines:
+  - name: query 
+    nodes:
+      - name: Retriever
+        inputs: [Query]
+```
+
+2. Create an Indexing Pipeline
+
+You can define an indexing pipeline in the same pipeline YAML file and index your documents to Elasticsearch through REST API. For that, create `FileTypeClassifier`, `TextConverter`, `PDFToTextConverter`, and last, `PreProcessor` nodes with parameters to split documents. 
+
+```yaml
+components:
+    ...
+  - name: FileTypeClassifier
+    type: FileTypeClassifier
+  - name: TextFileConverter
+    type: TextConverter
+  - name: PDFFileConverter
+    type: PDFToTextConverter
+  - name: Preprocessor
+    type: PreProcessor
+    params:
+      split_by: word
+      split_length: 1000
+      split_overlap: 50 
+      split_respect_sentence_boundary: True 
+```
+
+Then, go below to the `pipelines` section of the YAML file and create a new pipeline called `indexing`. In this pipeline, indicate how these nodes are connected to each other, Retriever, and DocumentStore. This indexing pipeline supports `.txt` and `.pdf` files and pre-processes them before loading to the Elasticsearch.
+
+```yaml
+pipelines:
+  ...
+  - name: indexing
+    nodes:
+      - name: FileTypeClassifier
+        inputs: [File]
+      - name: TextFileConverter
+        inputs: [FileTypeClassifier.output_1]
+      - name: PDFFileConverter
+        inputs: [FileTypeClassifier.output_2]
+      - name: Preprocessor
+        inputs: [PDFFileConverter, TextFileConverter]
+      - name: Retriever
+        inputs: [Preprocessor]
+      - name: DocumentStore
+        inputs: [Retriever]
+```
+
+After completing query and indexing pipelines, add `version: ignore` to the top of the file and the pipeline YAML file is ready to run by Haystack API.
+
+```yaml
+version: ignore
+
+components:
+  - name: DocumentStore
+    type: ElasticsearchDocumentStore
+  - name: Retriever
+    type: EmbeddingRetriever
+    params:
+      document_store: DocumentStore
+      top_k: 5 
```

Comment:
This is up to you, but I think the section just under these yaml definitions are a great opportunity to hint to the learner how they can change setting. It might serve as a good mental map from the API reference they can see in docs to implementing them in a YAML file. `top_k` for example but you can in a short paragraph maybe hint them into some other things they could set?

I'll leave this decision up to you and @brandenchan depending on the level of 'optionality' you decide to provide in this tutorial

Reply:
Good idea to mention API reference 💡  I don't want to offer any options such as saying that "add `max_seq_len` to Retriever", instead, encourage them to play with the pipeline later and refer to the API Reference for that. So, this sentence👇 might come after sharing the whole YAML file. WDYT? @TuanaCelik @brandenchan  

> Feel free to play with the pipeline setup later on. Add or remove some nodes, change the parameters or add new ones. Check out [Haystack API Reference](https://docs.haystack.deepset.ai/reference/answer-generator-api) for more options on nodes and parameters.